SharedClasses compile error on mac osx

Compiling on mac osx shows the following SCC error due to overloaded constructors with different parameters.  (One is missing the J9JavaVM* parameter)

```
In file included from OSCache.cpp:40:
./OSCachevmem.hpp:63:15: error: 'SH_OSCachevmem::startup' hides overloaded virtual function [-Werror,-Woverloaded-virtual]
        virtual bool startup(const char* ctrlDirName, UDATA cacheDirPerm, const char* cacheName, J9SharedClassPreinitConfig* piconfig, IDATA numLocks,
                     ^
./OSCache.hpp:204:15: note: hidden overloaded virtual function 'SH_OSCache::startup' declared here: different number of parameters (14 vs 13)
        virtual bool startup(J9JavaVM* vm, const char* cacheDirName, UDATA cacheDirPerm, const char* cacheName, J9SharedClassPreinitConfig* piconfig_, IDATA numLocks,
                     ^
```

For the mac build, I'm probably going to hack around it but we should look at what the right answer is.

--- a/runtime/shared_common/OSCache.cpp
+++ b/runtime/shared_common/OSCache.cpp
@@ -38,7 +38,9 @@
 
 #include "OSCachesysv.hpp"
 #include "OSCachemmap.hpp"
+#if defined(J9SHR_CACHELET_SUPPORT)
 #include "OSCachevmem.hpp"
+#endif /* J9SHR_CACHELET_SUPPORT */
 #include "CacheMap.hpp"
 
 /**

--- a/runtime/shared_common/ScopeManagerImpl.cpp
+++ b/runtime/shared_common/ScopeManagerImpl.cpp
@@ -1,5 +1,5 @@
 /*******************************************************************************
- * Copyright (c) 2001, 2014 IBM Corp. and others
+ * Copyright (c) 2001, 2018 IBM Corp. and others
  *
  * This program and the accompanying materials are made available under
  * the terms of the Eclipse Public License 2.0 which accompanies this
@@ -31,8 +31,10 @@
 #include "j9consts.h"
 #include <string.h>
 
-SH_ScopeManagerImpl::SH_ScopeManagerImpl():
-	_allCacheletsStarted(false)
+SH_ScopeManagerImpl::SH_ScopeManagerImpl()
+#if defined(J9SHR_CACHELET_SUPPORT)
+	:_allCacheletsStarted(false)
+#endif /* J9SHR_CACHELET_SUPPORT */
 {
 	_htMutexName = "scTableMutex";
 }

--- a/runtime/shared_common/ScopeManagerImpl.hpp
+++ b/runtime/shared_common/ScopeManagerImpl.hpp
@@ -1,5 +1,5 @@
 /*******************************************************************************
- * Copyright (c) 2001, 2017 IBM Corp. and others
+ * Copyright (c) 2001, 2018 IBM Corp. and others
  *
  * This program and the accompanying materials are made available under
  * the terms of the Eclipse Public License 2.0 which accompanies this
@@ -103,9 +103,9 @@ private:
 	IDATA scCollectHashes(J9VMThread* currentThread, SH_CompositeCache* cachelet, CacheletHints* hints);
 	static UDATA scCollectHashOfEntry(void* entry, void* userData);
 	static UDATA scCountCacheletHashes(void* entry, void* userData);
-#endif
-	
+
 	bool _allCacheletsStarted;
+#endif /* J9SHR_CACHELET_SUPPORT */
 };
 
 #endif /* SCOPEMANAGERIMPL_H_INCLUDED */

--- a/runtime/shared_common/shrinit.cpp
+++ b/runtime/shared_common/shrinit.cpp
@@ -1381,7 +1381,7 @@ hookFindSharedClass(J9HookInterface** hookInterface, UDATA eventNum, void* voidD
 		UDATA pathEntryCount = eventData->entryCount;
 
 		if (J2SE_VERSION(vm) >= J2SE_19) {
-			if ((eventData->classloader == vm->systemClassLoader)) {
+			if (eventData->classloader == vm->systemClassLoader) {
 				isBootLoader = true;
 				pathEntryCount += 1;
 				classpath = getBootstrapClasspathItem(currentThread, vm->modulesPathEntry, pathEntryCount);

